File: clients/javascript/src/models/msgQueueNackIn.ts
// this file is @generated

export interface MsgQueueNackIn {
    msgIds: string[];
}

export interface MsgQueueNackIn_ {
    topic: string;
    consumerGroup: string;
    msgIds: string[];
}

export const MsgQueueNackInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgQueueNackIn_ {
        return {
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            msgIds: object['msg_ids'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgQueueNackIn_): any {
        return {
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'msg_ids': self.msgIds,
        };
    }
}

File: z-clients/python/diom/models/msg_queue_nack_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgQueueNackIn(BaseModel):
    namespace: str | None = None

    msg_ids: t.List[str]


class _MsgQueueNackIn(BaseModel):
    namespace: str | None = None

    topic: str

    consumer_group: str

    msg_ids: t.List[str]
